### PR TITLE
add maxlength props for description > 255 chars and basic form validation, Fixes - 1342

### PR DESCRIPTION
**Pull request by maddhruv**
### Comprises -
- form validation
- error mounting fix
- the form on change reset error fix
- max length in input fields
for *Create Collective Form*

#### Fixes
https://github.com/opencollective/opencollective/issues/1342

### CODE REVIEW

**Comment by maddhruv** [comment]
@znarf Please have a look into

**Comment by maddhruv** [comment]
> start the frontend linked to the local API

how do I do that?
which npm script?

**Comment by znarf** [comment]
It's in the Readme.

> In case you want to connect to the Open Collective API running locally:
> clone, install and start opencollective-api
> in this project (frontend), copy `.env.local` to `.env`.

But at this point, there is no real need to run the e2e test locally, you can just make sure that the `/organizations/new` page works and it should be fine.

**Comment by maddhruv** [comment]
@znarf This look to be working fine now
deployed @ - https://opencollective-frontend-rinutknaiq.now.sh/organizations/new

**Comment by maddhruv** [comment]
No it is not happening for me 😕

**Comment by maddhruv** [comment]
@znarf 
please check again
my bad

**Comment by znarf** [comment]
If you're interested in e2e tests, I'm currently trying to improve the documentation, see https://github.com/opencollective/opencollective-frontend/pull/2616

File: components/CreateCollective.js [hunk 1]
```diff
@@ -78,16 +78,27 @@ class CreateCollective extends React.Component {
...
       return;
-    }
-    if (get(this.host, 'settings.tos') && !CollectiveInputType.hostTos) {
-      this.setState({
-        result: { error: 'Please accept the terms of fiscal sponsorship' },
-      });
+    } else if (!CollectiveInputType.description) {
+      this.error('Please enter a description');
+      return;
+    } else if (!CollectiveInputType.website) {
```
[the remark this answers is not in the document]
Comment (maddhruv) [reply components/CreateCollective.js]: I thought these are required fields

File: components/CreateCollective.js [hunk 2]
```diff
@@ -223,6 +234,7 @@ class CreateCollective extends React.Component {
                   collective={this.state.collective}
                   onSubmit={this.createCollective}
                   onChange={this.resetError}
+                  showForm={this.host.isHost ? true : false}
```
[the remark this answers is not in the document]
Comment (maddhruv) [reply components/CreateCollective.js]: having control over when to show the form

[the remark this answers is not in the document]
Comment (Betree) [reply components/CreateCollective.js]: What's the behavior change that requires us to have control over when to show the form? Why didn't we need this before?

File: components/CreateCollective.js [hunk 3]
```diff
@@ -78,20 +79,10 @@ class CreateCollective extends React.Component {
...
-      return;
-    } else if (!CollectiveInputType.website) {
-      this.error('Please enter a website url');
-      return;
-    } else if (CollectiveInputType.description.length > 255) {
+    if (CollectiveInputType.description.length > 255) {
       this.error('The description should be of less than 255 characters');
-    } else if (!websiteRegex.test(CollectiveInputType.website)) {
+      return;
+    } else if (!isURL(CollectiveInputType.website)) {
```
Comment (Betree) [inline components/CreateCollective.js]: From my understanding `isURL('') === false` (see https://codesandbox.io/s/icy-moon-f81we) so the form will be considered as invalid if we don't set a URL.

**Comment by Betree** [comment]
Also the two feedback items from https://github.com/opencollective/opencollective-frontend/pull/2605#pullrequestreview-293567269 have not been resolved nor replied to.

**Comment by maddhruv** [comment]
closing in favour of https://github.com/opencollective/opencollective-frontend/pull/2662

**Comment by znarf** [comment]
@maddhruv it's a bit sad to see some improvements you wanted to contribute gone in the new PR.

**Comment by maddhruv** [comment]
'cause I still don't have context with what all field values are required or so
and some other points about the flow

and didn't want to have a PR containing unrelated changes for a specific issue

yes some of the changes were good to have (validation and error handling)
but that I can eventually do in separate PRs

**Comment by Betree** [comment]
@maddhruv Which PR should we keep, this one or https://github.com/opencollective/opencollective-frontend/pull/2662?

**Comment by maddhruv** [comment]
> Which PR should we keep

if you want the validation and other minor pieces of stuff to land than **that** one
or if you just want to land the error fix than **this** one
😆

**Comment by Betree** [comment]
Ok, there are still un(resolved/responded) review items on this one tho

**Comment by maddhruv** [comment]
@Betree 
Added
- name field required (why? 'cause name field is required to create a collective)
- the valid website URL check (if it doesn't exist, don't check)

Removed
- unnecessary check on description length

File: components/CreateCollective.js [hunk 4]
```diff
@@ -78,16 +79,17 @@ class CreateCollective extends React.Component {
...
-      });
+    if (!CollectiveInputType.name) {
+      this.error('The name field is required');
       return;
-    }
-    if (get(this.host, 'settings.tos') && !CollectiveInputType.hostTos) {
-      this.setState({
-        result: { error: 'Please accept the terms of fiscal sponsorship' },
-      });
+    } else if (!!CollectiveInputType.website && !isURL(CollectiveInputType.website)) {
```
Comment (Betree) [inline components/CreateCollective.js]: Simpler is better

```suggestion
    } else if (CollectiveInputType.website && !isURL(CollectiveInputType.website)) {
```

File: components/CreateCollective.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Betree) [reply components/CreateCollective.js]: Still need an answer on this one :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (Betree) [reply components/CreateCollective.js]: Still need an answer on this one :slightly_smiling_face:

**Comment by maddhruv** [comment]
> add maxlength props for description > 255 chars

by that I meant
the max-length is 255 chars

**Comment by Betree** [comment]
> by that I meant the max-length is 255 chars

Ok but then why did you remove the description length check in https://github.com/opencollective/opencollective-frontend/pull/2605/commits/71ad70f1283daf1c6bc32e8e2685aca4a15f6f15?

**Comment by maddhruv** [comment]
I have already removed that 
https://github.com/opencollective/opencollective-frontend/pull/2605/files#diff-282fb4fabf50caed32c1a2b327825cd3R81

**Comment by maddhruv** [comment]
I even don't remember at what stage I added that state 😢 
We shall not go with these changes
Closing the PR

**Pull request closed without merging**